### PR TITLE
#164269844 Complete view and edit profiles

### PR DESCRIPTION
**Pull request by olajide1234**
#### What does this PR do?
It sets up routes and controllers for users to be able to view and edit their profiles.

#### Description of Task to be completed?
Users should be able to view and edit their profiles on Authors Haven.

#### How should this be manually tested?
- Run `npm run start` to start the application.
- Make a GET request to `/api/v1/userprofile` with the authentication token. Application will return basic profile.
- Make a PATCH request to `/api/v1/userprofile` with a JSON object containing all or some of the following fields: `firstname`, `lastname`, `bio`. Application will return updated profile.

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#164269844
#### Questions:
N/A

### CODE REVIEW

**Review by Dubby20 — COMMENTED** [review]
- Remove inline comment on line 4

File: controllers/profile.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+   * @memberof ProfileController
+   *
+   * @returns {object} User Profile
+   */
+  static async editProfile(req, res) {
+    const { id } = req.user;
+
+    try {
+      const updateResult = await User.update(
+        { bio: req.body.bio, name: `${req.body.firstname} ${req.body.lastname}` },
```
Comment (codinger41) [inline controllers/profile.js]: Destructure bio, firstname and lastname from req.body.

File: test/helpers/updateVerifiedStatus.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { User } from '../../models';
+
+export default async (username) => {
+  try {
+    await User.update({ verified: true }, { where: { username } });
+  } catch (error) {
+    return error;
+  }
+};
```
Comment (codinger41) [inline test/helpers/updateVerifiedStatus.js]: Fix EOL

**Review by Dubby20 — APPROVED** [review]
Looks good to merge

**Review by codinger41 — APPROVED** [review]

File: controllers/profile.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        }
+      });
+      delete foundUser.password;
+      delete foundUser.verificationId;
+
+      const splitNamesObject = splitName(foundUser);
+
+      return res.status(200).json({
+        success: true,
+        body: _.extendOwn(foundUser, splitNamesObject)
```
Comment (9jaswag) [inline controllers/profile.js]: @olajide1234  what does this do?

Comment (olajide1234) [reply controllers/profile.js]: It returns a new object which is a combination of the `foundUser` and `splitNamesObject` objects. https://underscorejs.org/#extendOwn

File: test/helpers/dummyData.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+export const helperUser = {
+  username: 'AyinlaOlajide',
+  email: 'helperuser@gmail.com',
+  name: 'Ayinla Olajide',
+  password: '1234567'
+};
+
+export const profileDetails = {
```
Comment (9jaswag) [inline test/helpers/dummyData.js]: @olajide1234 let's have different files for different dummy data. One for `user` related data, another for `articles` etc.
Once that is done, this is ready for merge

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: test/helpers/userDummyData.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+export const user1 = {
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply test/helpers/userDummyData.js:1]: Resolved, prettier setting history has been removed.

File: test/helpers/articleDummyData.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const article1 = {
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply test/helpers/articleDummyData.js:1]: Resolved, prettier setting history has been removed.

File: test/helpers/dummyData.js (hunk 4 above)
Comment (olajide1234) [reply test/helpers/dummyData.js]: Implemented as requested

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**